UserTools/EventSelector/EventSelector.cpp

```diff
@@ -109,6 +112,15 @@ bool EventSelector::Execute(){
   	Log("EventSelector  Tool: Error retrieving RecoDigits,no digit from the RecoEvent!",v_warning,verbosity); 
   	/*return false;*/
   }
+      auto has_LAPPDMult = m_data->Stores.at("RecoEvent")->Get("HitLAPPDs", fHitLAPPDs);
+      if (not has_LAPPDMult) {
+          Log("EventSelector Tool: No count of hit LAPPDs; proceeding without LAPPD multiplicity cut.", v_warning, verbosity);
+          fLAPPDMultMin = 0;
+          fHitLAPPDs = new std::vector<int>;
```

Comment:
This Tool doesn't seem to call `m_data->Stores.at("RecoEvent")->Set("HitLAPPDs",XXX)`, which means if `HitLAPPDs` doesn't exist for the first Execute call, it probably won't exist for any of them, and you'll be creating a new `std::vector<int>` on every loop, leaking them as you go.
Is there any reason this vector even needs to be on the heap? At the least please only `new` the vector if it doesn't exist.